src/backend/distributed/commands/function.c

```diff
@@ -70,10 +107,297 @@ create_distributed_function(PG_FUNCTION_ARGS)
 
 	MarkObjectDistributed(&functionAddress);
 
+	UpdateFunctionDistributionInfo(&functionAddress, distributionArgumentIndex,
+								   colocationId);
+
 	PG_RETURN_VOID();
 }
 
 
+/*
+ * GetDistributionArgIndex calculates the distribution argument with the given
+ * parameters. The function errors out if no valid argument is found.
+ */
+static int
+GetDistributionArgIndex(Oid functionOid, text *distributionArgumentNameText,
+						int distributionArgumentIndexInput, Oid *distributionArgumentOid)
+{
+	char *distributionArgumentName = text_to_cstring(distributionArgumentNameText);
+	int distributionArgumentIndex = -1;
+
+	int numberOfArgs = 0;
+	int argIndex = 0;
+	Oid *argTypes = NULL;
+	char **argNames = NULL;
+	char *argModes = NULL;
+
+	HeapTuple proctup = NULL;
+
+	*distributionArgumentOid = InvalidOid;
+
+	if (strlen(distributionArgumentName) == 0 && distributionArgumentIndexInput == -1)
+	{
+		char *functionName = get_func_name(functionOid);
+
+		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						errmsg("cannot distribute the function \"%s\" since "
+							   "the distribution argument is not provided",
+							   functionName),
+						errhint("Either provide distribution_arg_name or "
+								"distribution_arg_index to "
+								"create_distributed_function()")));
+	}
+	else if (strlen(distributionArgumentName) != 0 &&
+			 distributionArgumentIndexInput != -1)
+	{
+		char *functionName = get_func_name(functionOid);
+
+		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						errmsg("cannot distribute the function \"%s\" since "
+							   "multiple distribution arguments are provided",
+							   functionName),
+						errhint("Either provide distribution_arg_name or "
+								"distribution_arg_index to "
+								"create_distributed_function()")));
+	}
+
+	proctup = SearchSysCache1(PROCOID, ObjectIdGetDatum(functionOid));
+	if (!HeapTupleIsValid(proctup))
+	{
+		elog(ERROR, "cache lookup failed for function %u", functionOid);
+	}
+
+	numberOfArgs = get_func_arg_info(proctup, &argTypes, &argNames, &argModes);
+
+	if (distributionArgumentIndexInput != -1 &&
+		(distributionArgumentIndexInput < 1 ||
+		 distributionArgumentIndexInput > numberOfArgs))
+	{
+		char *functionName = get_func_name(functionOid);
+
+		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						errmsg("cannot distribute the function \"%s\" since the "
+							   "distribution argument is not valid ", functionName),
+						errhint("Provide a valid distribution_arg_index to "
+								"create_distributed_function()")));
+	}
+	else if (distributionArgumentIndexInput != -1)
+	{
+		/*
+		 * Internal representation for the distributionArgumentIndex
+		 * starts from 0 whereas user facing API starts from 1.
```

Comment:
Why do we start the user-facing one from 1?

Reply:
I couldn't find any Citus UDF that asks for an index, do you remember any?
Picked this mostly because arrays starts from `1` in Postgres, but I can change it back if it feels confusing.

Reply:
I think, with the new commit, starting from `1` becomes more intuitive because `$1` is the way to access the first parameter in the functions.